modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -4827,6 +4934,10 @@ private void processNodeAddFinishedMessage(TcpDiscoveryNodeAddFinishedMessage ms
             if (log.isDebugEnabled())
                 log.debug("Node to finish add: " + node);
 
+            //we will need to recalculate this value since the topology changed
+            clusterSupportsTcpDiscoveryNodeSerializationOptimization &=
```

Comment:
No need to recalculate this property if node joins to the cluster that doesn't support the feature.
New node supporting the feature won't change the flag if there are nodes not supporting it.